tools/.shared-functions.sh

```diff
@@ -44,3 +43,20 @@ get_css_file()
     debug "  css file saved ${css_file}"
     echo "${css_file}"
 }
+
+
+check_xcode() {
+  if [[ $(uname) == "Darwin" ]]; then
+    xcode_version=$(xcodebuild -version | grep "Xcode" | awk '{print $2}' | sed 's/[.].*//')
+    minimum_xcode_version=12
+    if [[ $(printf '%s\n' "$xcode_version" "$minimum_xcode_version" | sort -V | head -n1) == "$xcode_version" ]]; then
+      echo
```

Comment:
According to the description of the issue, I think anything other than XCode 12 should fail here, right?
1. maybe `minimum_xcode_version` should be something more like `supported_xcode_version`?
2. This test as written passes with XCode 13 & 14, which it should not, correct?